src/App.tsx

```diff
@@ -1291,19 +1339,146 @@ function App() {
     [copiedFilePaths, currentFolderPath, handleLibraryRefresh],
   );
 
-  const handleZoomChange = useCallback((newZoomValue: number) => {
+  // Request full resolution preview when zoomed in
+  const requestFullResolution = useCallback(
+    debounce((currentAdjustments: any) => {
+      if (!selectedImage?.path) return;
+      
+      if (fullResRequestRef.current) {
+        fullResRequestRef.current.cancelled = true;
+      }
+      
+      const request = { cancelled: false };
+      fullResRequestRef.current = request;
+      
+      invoke('generate_fullscreen_preview', {
+        jsAdjustments: currentAdjustments
+      }).then((fullResUrl: string) => {
+        if (!request.cancelled) {
+          setFullResolutionUrl(fullResUrl);
+          setIsFullResolution(true);
+          setIsLoadingFullRes(false);
+        }
+      }).catch((error: any) => {
+        if (!request.cancelled) {
+          console.error('Failed to generate full resolution preview:', error);
+          setIsFullResolution(false);
+          setFullResolutionUrl(null);
+          setIsLoadingFullRes(false);
+        }
+      });
+    }, 300),
+    [selectedImage?.path]
+  );
+  
+  // Regenerate full resolution when adjustments change while zoomed
+  useEffect(() => {
+    if (isFullResolution && selectedImage?.path) {
+      setFullResolutionUrl(null);
+      setIsLoadingFullRes(true);
+      requestFullResolution(adjustments);
+    }
+  }, [adjustments, isFullResolution, selectedImage?.path, requestFullResolution]);
+
+  // Handle zoom changes with percentage relative to original image size
+  const handleZoomChange = useCallback((input: number | string) => {
```

Comment:
Do not use `number | string` as variable type. Use either one. A good example is the code above which doesn't make any sense.
```
let targetZoomPercent = input;
if (input === 'fit-to-window') {
```